Mathlib/Probability/Distributions/Uniform.lean

```diff
@@ -0,0 +1,217 @@
+/-
+Copyright (c) 2024 Josha Dekker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Josha Dekker
+-/
+import Mathlib.Probability.Notation
+import Mathlib.Probability.Cdf
+import Mathlib.Probability.Density
+
+/-! # Uniform distributions
+
+Defines the uniform distribution for any set with finite volume.
+
+## Main definitions
+* `uniformVolume s` : The uniform volume on `s` is the
+  the volume measure restricted to `s`, normalized.
+* `IsUniform X s ℙ μ` : A random variable `X` has uniform distribution on `s` under `ℙ` if the
+  push-forward measure agrees with the rescaled restricted volume measure `μ`.
+-/
+
+open scoped Classical MeasureTheory NNReal ENNReal
+
+open TopologicalSpace MeasureTheory.Measure
+
+noncomputable section
+
+namespace MeasureTheory
+
+variable {E : Type*} [MeasurableSpace E] {m : Measure E} {μ : Measure E}
+
+/-- A measure is a uniform volume for a set `s` if it is the rescaled restriction of the volume to
+this set.  -/
+def uniformVolume (s : Set E) (μ : Measure E := by volume_tac) : Measure E := (μ s)⁻¹ • μ.restrict s
+
+namespace UniformVolume
+
+theorem absolutelyContinuous {s : Set E} :
+    uniformVolume s μ ≪ μ := by
+  intro t ht
+  unfold uniformVolume
+  rw [smul_apply, smul_eq_mul]
+  apply mul_eq_zero.mpr
+  refine Or.inr (le_antisymm ?_ (zero_le _))
+  exact ht ▸ restrict_apply_le s t
+
+theorem measure_preimage {s : Set E} {A : Set E}
+    (hA : MeasurableSet A) (hunif : m = uniformVolume s μ): m A = μ (s ∩ A) / μ s := by
```

Comment:
`m` is useless here. Simply replace it with `uniformVolume s μ` in the statement. Also I don't understand the name: there is no preimage involved.

You could remove that lemma from the namespace UniformVolume (and consider removing all lemmas from that namespace) and rename it `uniformVolume_apply`.

Reply:
Thank you for the feedback! I'm not sure what you mean by "removing all lemmas from the namespace UniformVolume": you mean that I should get rid of the lemmas, or just move them up one level?